### PR TITLE
Filestorage roadmap

### PR DESCRIPTION
**Pull request by diegohce**
clean commits

### CODE REVIEW

**Comment by deadprogram** [comment]
Thank you very much for the cleanup here @diegohce 

Now we probably just need to figure out the permissions on Windows for file writing. Excellent progress, I will try to take a look sometime today.

**Comment by deadprogram** [comment]
Please note I changed the branch for this PR to `dev`. :smile_cat:

**Comment by diegohce** [comment]
> Please note I changed the branch for this PR to `dev`. 😸

Did I mistake the branch... `again!` ?

**Comment by diegohce** [comment]
Damn! I'm so eager for this to get to the new release!

File: persistence_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package gocv
+
+import (
+	"testing"
+)
+
+func TestFileStorage(t *testing.T) {
+
+	fs := NewFileStorageWithParams("testdata/filestorage", FileStorageModeWrite|FileStorageModeFormatJson, "utf-8")
```
Comment (deadprogram) [inline persistence_test.go]: Perhaps writing the file into a temp directory would be better and would pass on Windows?

Untested:

```go
pth := filepath.Join(os.TempDir(), "filestorage.json")
fs := NewFileStorageWithParams(pth, FileStorageModeWrite|FileStorageModeFormatJson, "utf-8")
```

**Comment by diegohce** [comment]
Would "/tmp" be a factible directory?

**Comment by diegohce** [comment]
sorry, I missed your code sample. I will try with that right now

File: persistence_filestorage.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	defer C.free(unsafe.Pointer(c_val))
+
+	C.FileStorage_WriteString(fs.p, c_name, c_val)
+}
+
+func (fs *FileStorage) WriteStringArray(name string, val []string) {
+	c_name := C.CString(name)
+	defer C.free(unsafe.Pointer(c_name))
+
+	//c_val := []*C.char{}
```
Comment (deadprogram) [inline persistence_filestorage.go]: Comment should be removed, no?

File: persistence_filestorage.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+	c_val := make([]*C.char, 0, len(val))
+
+	for _, v := range val {
+		c_val = append(c_val, C.CString(v))
+	}
+	defer func() {
+		for _, p := range c_val {
+			C.free(unsafe.Pointer(p))
+		}
+		//C.free(unsafe.Pointer(c_val))
```
Comment (deadprogram) [inline persistence_filestorage.go]: Should this also be removed?

File: persistence_filestorage.go (hunk 2 above)
Comment (diegohce) [reply persistence_filestorage.go]: yes

File: persistence_filestorage.go (hunk 3 above)
Comment (diegohce) [reply persistence_filestorage.go]: yes. I will remove both and PR again

File: persistence_test.go (hunk 1 above)
Comment (diegohce) [reply persistence_test.go]: change done

**Comment by deadprogram** [comment]
Looking good now @diegohce thank you very much for adding this set of long-needed features! Now squash/merging.

**Pull request merged**